features/nvstore/source/nvstore.h

```diff
@@ -148,6 +158,24 @@ class NVStore : private mbed::NonCopyable<NVStore> {
      */
     int set(uint16_t key, uint16_t buf_size, const void *buf);
 
+    /**
+     * @brief Programs one item of data on Flash, allocating a key.
+     *
+     * @param[out] key                  Returned key of stored item.
+     * @param[in]  buf_size             Item size in bytes.
+     * @param[in]  buf                  Buffer containing data.
+     *
+     * @returns NVSTORE_SUCCESS           Value was successfully written on Flash.
+     *          NVSTORE_WRITE_ERROR       Physical error writing data.
+     *          NVSTORE_BAD_VALUE         Bad value in any of the parameters.
+     *          NVSTORE_FLASH_AREA_TOO_SMALL
+     *                                    Not enough space in Flash area.
+     *          NVSTORE_ALREADY_EXISTS    Item set with write once API already exists.
+     *          NVSTORE_NO_FREE_KEY       Couldn't allocate a key for this call.
+     *
+     */
+    int set_alloc_key(uint16_t &key, uint16_t buf_size, const void *buf);
```

Comment:
The mbed convention is to use pointers instead of references. It's easier to tell if the function modifies the value at the call site.

Reply:
Actually nevermind me, this matches the rest of the implementation.